libs/entitlement/qa/entitlement-test-plugin/src/main/java/org/elasticsearch/entitlement/qa/test/RestEntitlementsCheckAction.java

```diff
@@ -49,27 +48,34 @@ record CheckAction(
         Integer fromJavaVersion
     ) {}
 
-    private static final Map<String, CheckAction> checkActions = Stream.of(
-        getTestEntries(FileCheckActions.class),
-        getTestEntries(FileStoreActions.class),
-        getTestEntries(JvmActions.class),
-        getTestEntries(LoadNativeLibrariesCheckActions.class),
-        getTestEntries(ManageThreadsActions.class),
-        getTestEntries(NativeActions.class),
-        getTestEntries(NetworkAccessCheckActions.class),
-        getTestEntries(NioChannelsActions.class),
-        getTestEntries(NioFilesActions.class),
-        getTestEntries(NioFileSystemActions.class),
-        getTestEntries(OperatingSystemActions.class),
-        getTestEntries(PathActions.class),
-        getTestEntries(SpiActions.class),
-        getTestEntries(SystemActions.class),
-        getTestEntries(URLConnectionFileActions.class),
-        getTestEntries(URLConnectionNetworkActions.class)
-    )
-        .flatMap(Function.identity())
-        .filter(entry -> entry.getValue().fromJavaVersion() == null || Runtime.version().feature() >= entry.getValue().fromJavaVersion())
-        .collect(Collectors.toUnmodifiableMap(Entry::getKey, Entry::getValue));
+    private static final Map<String, CheckAction> checkActions = collectTests(
+        FileCheckActions.class,
+        FileStoreActions.class,
+        JvmActions.class,
+        LoadNativeLibrariesCheckActions.class,
+        ManageThreadsActions.class,
+        NativeActions.class,
+        NetworkAccessCheckActions.class,
+        NioChannelsActions.class,
+        NioFilesActions.class,
+        NioFileSystemActions.class,
+        OperatingSystemActions.class,
+        PathActions.class,
+        SpiActions.class,
+        SystemActions.class,
+        URLConnectionFileActions.class,
+        URLConnectionNetworkActions.class
+    );
+
+    private static Map<String, CheckAction> collectTests(Class<?>... testClasses) {
+        List<Entry<String, CheckAction>> entries = new ArrayList<>();
+        for (Class<?> testClass : testClasses) {
+            getTestEntries(entries, testClass, a -> a.fromJavaVersion() == null || Runtime.version().feature() >= a.fromJavaVersion());
+        }
+        @SuppressWarnings("unchecked")
+        Entry<String, CheckAction>[] entriesArray = entries.toArray(new Entry[0]);
```

Comment:
```
warning: [rawtypes] found raw type: Entry
--
  | Entry<String, CheckAction>[] entriesArray = entries.toArray(new Entry[0]);
```

Reply:
Oops. I've brought this PR up to date now and fixed this warning, see [647f97d](https://github.com/elastic/elasticsearch/pull/125449/commits/647f97dbd2f8ab422497cb3966191de530406a0a)